lib/vdsm/virt/vmdevices/storage.py

```diff
@@ -940,6 +940,26 @@ def on_block_threshold(self, reported_path):
             self.exceeded_time = time.monotonic_time()
             self.log.info("drive %r threshold exceeded", self.name)
 
+    def on_enospc(self):
+        """
+        Called when a VM pause because of ENOSPC writing to this drive.
+
+        We mark the drive for extension so it will be picked up by the priodic
+        monitor later.
```

Comment:
Here would be IMHO useful to add reasoning from the commit message that we don't have to hurry up as VM is already paused (as the natural question when reading this comment would probably be why not trigger extension immediately from here).

Reply:
ack